### PR TITLE
[PROCS-3943] Add status for process agent component on core agent

### PR DESCRIPTION
**Pull request by daniel-taf**
### What does this PR do?

This PR adds the status of process-agent component to the `datadog-agent status` output. This provides insight into what process checks are running on the core agent and similar stats to what `process-agent status` displays.

### Motivation

PROCS-3943

### Additional Notes

I had considered creating a separate status provider component decoupled from the process agent component, however I felt it was easier to manage its use by tying it to the process agent component.

Also, as this implementation is fairly similar to the existing one for the [agent](https://github.com/DataDog/datadog-agent/blob/main/comp/process/status/statusimpl/status.go), I had considered re-using the component and adding some behavior to switch the expvar server used based on the agent flavor. However, that would not be feasible if we want to show both component and agent binary status as different sections like now.

### Describe how to test/QA your changes

There are three scenarios to test:
- process checks run on Core Agent + Process Agent disabled
- process checks run on Core Agent + Process Agent enabled
- Process Agent enabled

**Process Agent enabled**
Using these configs:
```
// datadog.yaml
process_config:
  process_collection:
    enabled: true
```

Verify both status commands display the same result as before:
 * `datadog-agent status`
 *  `process-agent status`

**process checks run on Core Agent + Process Agent disabled**
Using these configs:
```
// datadog.yaml
process_config:
  process_collection:
    enabled: true
  run_in_core_agent:
    enabled: true
```

* Verify the `datadog-agent status` only shows the new process component status
```
=============
Process Agent
=============

  Status: Not running or unreachable

=================
Process Component
=================


  Enabled Checks: [process rtprocess]
  System Probe Process Module Status: Not running
  Process Language Detection Enabled: False

  =================
  Process Endpoints
  =================
    https://process.datadoghq.com - API Key ending with:
        - f589f

  =========
  Collector
  =========
    Last collection time: 2024-04-11 15:03:23
    Docker socket:
    Number of processes: 0
    Number of containers: 0
    Process Queue length: 0
    RTProcess Queue length: 0
    Connections Queue length: 0
    Event Queue length: 0
    Pod Queue length: 0
    Process Bytes enqueued: 0
    RTProcess Bytes enqueued: 0
    Connections Bytes enqueued: 0
    Event Bytes enqueued: 0
    Pod Bytes enqueued: 0
    Drop Check Payloads: []

  ==========
  Extractors
  ==========

    Workloadmeta
    ============
      Cache size: 0
      Stale diffs discarded: 0
      Diffs dropped: 0
 ```

 **process checks run on Core Agent + Process Agent enabled**
Using these configs:
```
// system-probe.yaml
network_config:
  enabled: true
```
```
// datadog.yaml
process_config:
  process_collection:
    enabled: true
  run_in_core_agent:
    enabled: true
```
* Verify the `datadog-agent status` displays the process agent and process component status
* Verify the `process-agent status` displays only the process agent status

### CODE REVIEW

File: releasenotes/notes/process-component-status-66c1ce5c3a34831e.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Status of process agent component will be shown when used in the core Agent.
```
Comment (alai97) [inline releasenotes/notes/process-component-status-66c1ce5c3a34831e.yaml:11]:
```suggestion
    Status of the Process Agent component will be shown when used in the core Agent.
```

**Review by alai97 — APPROVED** [review]
Looks good, one suggestion.

File: comp/process/agent/status_test.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		} else {
+			_, err := w.Write(response)
+			require.NoError(t, err)
+		}
+	}
+
+	return httptest.NewServer(http.HandlerFunc(handler))
+}
+
+func TestStatus(t *testing.T) {
```
Comment (daniel-taf) [inline comp/process/agent/status_test.go:40]: Not sure if having this test is worthwhile. It's nearly the same as https://github.com/DataDog/datadog-agent/blob/main/comp/process/status/statusimpl/status_test.go. Might be worthwhile if the behavior diverges more in the future.

File: comp/process/expvars/expvarsimpl/expvars.go [hunk 3]
```diff
@@ -59,6 +60,11 @@ func newExpvarServer(deps dependencies) (expvars.Component, error) {
 		return nil, err
 	}
 
+	if flavor.GetFlavor() == flavor.DefaultAgent {
```
Comment (GustavoCaso) [inline comp/process/expvars/expvarsimpl/expvars.go:63]: Should we have the condition check that if not process agent return nil? 

```go
if flavor.GetFlavor() != flavor.ProcessAgent {
    return nil, nil
}
```

It reads better the intention that we only want to initialize the server if is the process agent process

File: comp/process/agent/agentimpl/agent.go [hunk 4]
```diff
@@ -81,7 +94,16 @@ func newProcessAgent(p processAgentParams) agent.Component {
...
+	if flavor.GetFlavor() == flavor.ProcessAgent {
+		return provides{
+			Comp: processAgentComponent,
+		}
+	}
+
+	return provides{
+		Comp:           processAgentComponent,
+		StatusProvider: statusComponent.NewInformationProvider(agent.NewStatusProvider(p.Config)),
+	}
```
Comment (GustavoCaso) [inline comp/process/agent/agentimpl/agent.go:106]: maybe checking if the agent flavor is not process agent read better 

```go
if flavor.GetFlavor() != flavor.ProcessAgent {
	return provides{
		Comp: processAgentComponent,
		StatusProvider: statusComponent.NewInformationProvider(agent.NewStatusProvider(p.Config))
	}
}

return provides{
	Comp:           processAgentComponent,
}
```

Also, I would add a comment explaining why in the case of not being the process agent we return a status provider

**Review by GustavoCaso — APPROVED** [review]
Code LGTM 👌 
Left a few suggestion comments but are not blocker. I consider the one about adding a comment to explain why the process agent might or might not include a status provider important as it is not clear from the code why we have that condition.

File: comp/process/agent/agentimpl/agent.go (hunk 4 above)
Comment (daniel-taf) [reply comp/process/agent/agentimpl/agent.go]: Done!

File: comp/process/expvars/expvarsimpl/expvars.go (hunk 3 above)
Comment (daniel-taf) [reply comp/process/expvars/expvarsimpl/expvars.go]: makes sense to me. Done

File: comp/process/agent/agentimpl/agent.go [hunk 5]
```diff
@@ -81,7 +94,16 @@ func newProcessAgent(p processAgentParams) agent.Component {
 		Log:     p.Log,
 	}
 
-	return processAgentComponent
+	if flavor.GetFlavor() != flavor.ProcessAgent {
```
Comment (ogaca-dd) [inline comp/process/agent/agentimpl/agent.go:97]: Code in `comp/process/agent` is specific to the process Agent, isn't it? When `flavor.GetFlavor()` is different from `flavor.ProcessAgent`?

File: comp/process/expvars/expvarsimpl/expvars.go [hunk 6]
```diff
@@ -59,6 +60,11 @@ func newExpvarServer(deps dependencies) (expvars.Component, error) {
 		return nil, err
 	}
 
+	if flavor.GetFlavor() != flavor.ProcessAgent {
+		// Don't run the server outside of the process agent
+		return nil, nil
```
Comment (ogaca-dd) [inline comp/process/expvars/expvarsimpl/expvars.go:65]: All components should be not nil. Returning `nil` here force the user of  `expvars.Component` to know that the component can be nil. 
We usually use optional.Option to make it explicit that the component is optional.

**Review by robertjli — APPROVED** [review]

File: comp/process/agent/agentimpl/agent.go (hunk 5 above)
Comment (daniel-taf) [reply comp/process/agent/agentimpl/agent.go:97]: Currently, we are in the process of migrating the process checks the core agent thus the component can actually be used in the core agent.

File: comp/process/agent/agentimpl/agent.go [hunk 7]
```diff
@@ -81,7 +101,20 @@ func newProcessAgent(p processAgentParams) agent.Component {
...
+		// We return a status provider when the component is used outside of the process agent
+		// as the component status is unique from the typical agent status in this case.
+		err := expvars.InitProcessStatus(p.Config, p.SysProbeConfig, p.HostInfo, p.Log, p.Telemetry)
+		if err != nil {
+			_ = deps.Log.Critical("Failed to initialize process status server:", err)
```
Comment (robertjli) [inline comp/process/agent/agentimpl/agent.go:109]: Should we rename the function param to `deps` to be consistent with other component implementations?

Comment (daniel-taf) [reply comp/process/agent/agentimpl/agent.go:109]: Yeah that makes sense. Its less to keep track of. Done

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**